wren-ai-service/src/providers/llm/ollama.py

```diff
@@ -126,182 +119,29 @@ async def run(
-        meta: Dict[str, Any] = {"model": self.model}
-
-        async with aiohttp.ClientSession(
-            timeout=aiohttp.ClientTimeout(self.timeout)
-        ) as session:
-            for i in tqdm(
-                range(0, len(texts_to_embed), batch_size),
-                disable=not self.progress_bar,
-                desc="Calculating embeddings",
-            ):
-                batch = texts_to_embed[i]  # Single batch only
-                payload = self._create_json_payload(batch, generation_kwargs)
-
-                async with session.post(
-                    self.url,
-                    json=payload,
-                ) as response:
-                    result = await response.json()
-                    all_embeddings.append(result["embedding"])
-
-        return all_embeddings, meta
-
-    @component.output_types(embedding=List[float], meta=Dict[str, Any])
-    async def run(
-        self,
-        documents: List[str],
-        generation_kwargs: Optional[Dict[str, Any]] = None,
-    ):
-        logger.debug(f"Running Ollama document embedder with documents: {documents}")
-
-        if (
-            not isinstance(documents, list)
-            or documents
-            and not isinstance(documents[0], Document)
-        ):
-            msg = (
-                "OllamaDocumentEmbedder expects a list of Documents as input."
-                "In case you want to embed a list of strings, please use the OllamaTextEmbedder."
-            )
-            raise TypeError(msg)
-
-        texts_to_embed = self._prepare_texts_to_embed(documents=documents)
-        embeddings, meta = await self._embed_batch(
-            texts_to_embed=texts_to_embed,
-            batch_size=self.batch_size,
-            generation_kwargs=generation_kwargs,
-        )
-
-        for doc, emb in zip(documents, embeddings):
-            doc.embedding = emb
-
-        return {"documents": documents, "meta": meta}
-
-
-@provider("ollama")
+@provider("ollama_llm")
 class OllamaLLMProvider(LLMProvider):
     def __init__(
         self,
-        url: str = os.getenv("OLLAMA_URL") or OLLAMA_URL,
-        generation_model: str = os.getenv("GENERATION_MODEL") or GENERATION_MODEL_NAME,
-        embedding_model: str = os.getenv("EMBEDDING_MODEL") or EMBEDDING_MODEL_NAME,
+        url: str = os.getenv("LLM_OLLAMA_URL") or LLM_OLLAMA_URL,
+        generation_model: str = os.getenv("GENERATION_MODEL") or GENERATION_MODEL,
     ):
-        logger.info(f"Using Ollama Generation Model: {generation_model}")
+        logger.info(f"Using Ollama LLM: {generation_model}")
         self._url = url
         self._generation_model = generation_model
-        self._embedding_model = embedding_model
 
     def get_generator(
         self,
-        model_kwargs: Optional[Dict[str, Any]] = GENERATION_MODEL_KWARGS,
+        model_kwargs: Dict[str, Any] = orjson.loads(
+            os.getenv("GENERATION_MODEL_KWARGS", "{}")
+        )
+        or GENERATION_MODEL_KWARGS,
```

Comment:
it is a gentle comment to mention here has the same situation that we mentioned in other comment! 😆